base/interactiveutil.jl

```diff
@@ -254,8 +254,14 @@ function gen_call_with_extracted_types(fcn, ex0)
                     Expr(:call, typesof, map(esc, args[2:end])...))
     end
     exret = Expr(:none)
+    is_macro = false
     ex = expand(ex0)
-    if !isa(ex, Expr)
+    if fcn in [:which, :less, :edit] && ex0.head == :macrocall
+        # special case @edit @time 1+2 to edit the macro
+        is_macro = true
```

Comment:
I understand your concern, but from my point of view it do not make sense to ask for `@edit @random_macro max(1,1)` and not point to the macro. As a macro can generate arbitrary code for you, and hence you do not know what you call `@edit` on.